### PR TITLE
fix(ui): TE-2112 rca heatmap page changes

### PR DESCRIPTION
**Pull request by alcatraz627**
#### Issue(s)

[TE-2112](https://startree.atlassian.net/browse/TE-2112)
[UX Mocks](https://www.figma.com/file/TJJC3cxS36JGSdsnPhsxTN/RCA-%2F-Re-design-%2F-Top-contributors-%2B-Heatmap?type=design&node-id=1605-58172&mode=design&t=pNVdRZwIMxuA5rOu-0)

#### Description

- Merge all the tabs for rca what-where page into a single screen
- Implement a top-level card and accordion for UI
- Enable sorting heatmap rows based on the order of top contributors
- Route changes to reflect the changes above
- Refactored file structure to reflect the above changes

#### Screenshots

https://github.com/startreedata/thirdeye/assets/20799363/25bf071b-c27a-459e-bdc1-3ebc16e3f05f

### CODE REVIEW

File: thirdeye-ui/src/app/components/rca/heat-map-section/heat-map-section.component.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import { Box, Button, Grid, Typography } from "@material-ui/core";
```
Comment (alcatraz627) [inline thirdeye-ui/src/app/components/rca/heat-map-section/heat-map-section.component.tsx:15]: This is the renamed https://github.com/startreedata/thirdeye/pull/1360/files#diff-1c98ced186fa5a374cca0cca6048442290b48804c52809d50519fe691e157581 file that is shown as deleted below

File: thirdeye-ui/src/app/components/rca/heat-map-section/heat-map-section.component.tsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    deserializeKeyValuePair,
+    serializeKeyValuePair,
+} from "../../../utils/params/params.util";
+import { RootCauseAnalysisForAnomalyPageParams } from "../../../pages/root-cause-analysis-for-anomaly-page/root-cause-analysis-for-anomaly-page.interfaces";
+import { InvestigationContext } from "../../../pages/rca/investigation-state-tracker-container-page/investigation-state-tracker.interfaces";
+import { HeatMapSectionProps } from "./heat-map-section.interfaces";
+
+const HEATMAP_FILTERS_URL_KEY = "heatmapFilters";
+
+export const HeatMapPage: FunctionComponent<HeatMapSectionProps> = ({
```
Comment (alcatraz627) [inline thirdeye-ui/src/app/components/rca/heat-map-section/heat-map-section.component.tsx:52]: This is the renamed [#1360 (files)](https://github.com/startreedata/thirdeye/pull/1360/files#diff-1c98ced186fa5a374cca0cca6048442290b48804c52809d50519fe691e157581) file that is shown as deleted below

File: thirdeye-ui/src/app/components/rca/top-contributors-section/top-contributors-section.component.tsx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+import { InvestigationContext } from "../../../pages/rca/investigation-state-tracker-container-page/investigation-state-tracker.interfaces";
+import { TopContributorsSectionProps } from "./top-contributors-section.interfaces";
+import {
+    Accordion,
+    AccordionDetails,
+    AccordionSummary,
+} from "@material-ui/core";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+
+export const TopContributorsSection: FunctionComponent<TopContributorsSectionProps> =
```
Comment (alcatraz627) [inline thirdeye-ui/src/app/components/rca/top-contributors-section/top-contributors-section.component.tsx:53]: This is the renamed [#1360 (files)](https://github.com/startreedata/thirdeye/pull/1360/files#diff-cf1432596f3100381e2821f310da3e719c13951c0730f31044cb48ccd4b0513c) file that is shown as deleted below

**Review by chouclee — APPROVED** [review]

File: thirdeye-ui/src/app/locale/languages/en-us.json [hunk 4]
```diff
@@ -218,7 +218,9 @@
         "has-feedback": "Has Feedback",
         "health-status": "Health Status",
         "heatmap": "Heatmap",
+        "heatmap-and-dimension-drills": "Heatmap & Dimension Drills",
         "heatmap-data": "Heatmap Data",
+        "heatmap-of-dimension-distribution": "Heatmap of Dimension distribution",
```
Comment (chouclee) [inline thirdeye-ui/src/app/locale/languages/en-us.json:223]: nit. `Heatmap of Dimension Distribution`? Not sure the strategy used in TE for capitalization vs non-capitalization.

**Pull request merged**